### PR TITLE
Add `NATGateway` CRD

### PR DESCRIPTION
**Pull request by RedbackThomson**
Issue #, if available:

Description of changes:
Creates the `NATGateway` CRD, with references to `ElasticIPAddress` and `Subnet` for their respective fields

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by jaypipes — APPROVED** [review]
Left a comment inline to address in a future PR. All the generated code and tests in here look A-OK, thank you @RedbackThomson :)

File: test/e2e/tests/__init__.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
```
Comment (brycahta) [inline test/e2e/tests/__init__.py:1]: what's the purpose of this file?

File: test/e2e/tests/helper.py [hunk 2]
```diff
@@ -36,6 +36,19 @@ def assert_internet_gateway(self, ig_id: str, exists=True):
             pass
         assert res_found is exists
 
+    def assert_nat_gateway(self, ngw_id: str, exists=True):
```
[the remark this answers is not in the document]
Comment (brycahta) [reply test/e2e/tests/helper.py:39]: https://github.com/aws-controllers-k8s/community/issues/1235

**Review by brycahta — APPROVED** [review]
Looks good

**Review by A-Hilaly — APPROVED** [review]
Neat!

**Review by brycahta — COMMENTED** [review]
/lgtm

**Pull request merged**